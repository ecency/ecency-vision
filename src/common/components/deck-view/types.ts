export interface DeckModel {
  data: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
  };
  listItemComponent: any;
  createdAt: Date;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}
