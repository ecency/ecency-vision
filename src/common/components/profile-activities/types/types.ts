export interface ActivityTypes {
    type: string;
      parent_author?: string;
      parent_permlink?: string;
      author?: string;
      permlink?: string;
      payment?: number;
      payer?: string;
      following?: string;
      community?: string;
      approve?: boolean;
      proposal_ids?: string;
      account?: string;
      id?: string;
      what?: string[];
      voter?: string;
      witness?: string;
      json: string;
    timestamp: string;
  }