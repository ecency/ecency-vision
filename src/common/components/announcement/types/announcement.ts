export interface Announcement {
  id: number;
  title: string;
  description: string;
  button_text: string;
  button_link: string;
}

export interface LaterAnnouncement {
  id: number;
  dateTime: Date;
}
