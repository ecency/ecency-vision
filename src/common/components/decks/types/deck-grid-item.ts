interface BaseDeckGridItem {
  id: string;
  key: number;
  type:
    | "u"
    | "ac"
    | "co"
    | "w"
    | "n"
    | "tr"
    | "to"
    | "s"
    | "cu"
    | "th"
    | "msf"
    | "faq"
    | "wb"
    | "wn";
  settings: Record<string, any>;
}

export interface ReloadableDeckGridItem extends BaseDeckGridItem {
  settings: {
    updateIntervalMs: number;
  };
}

export interface UserDeckGridItem extends ReloadableDeckGridItem {
  type: "u";
  settings: ReloadableDeckGridItem["settings"] & {
    username: string;
    contentType: string;
  };
}

export interface CommunityDeckGridItem extends ReloadableDeckGridItem {
  type: "co";
  settings: UserDeckGridItem["settings"] & {
    tag: string;
  };
}

export interface SearchDeckGridItem extends ReloadableDeckGridItem {
  type: "s";
  settings: ReloadableDeckGridItem["settings"] & {
    query: string;
    author?: string;
    type?: string;
    category?: string;
    tags?: string;
    date?: string;
    sort?: string;
    hideLow?: boolean;
  };
}

export interface WavesDeckGridItem extends ReloadableDeckGridItem {
  type: "th";
  settings: ReloadableDeckGridItem["settings"] & {
    host: string;
  };
}

export type DeckGridItem =
  | BaseDeckGridItem
  | UserDeckGridItem
  | CommunityDeckGridItem
  | SearchDeckGridItem
  | WavesDeckGridItem;
