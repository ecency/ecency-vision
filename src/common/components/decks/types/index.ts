export * from "./deck-grid-item";
export * from "./deck-grid";
export * from "./old-decks";
