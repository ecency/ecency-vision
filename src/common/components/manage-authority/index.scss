@import "src/style/vars_mixins";

.authority-table {
  table {
    .table-head {
      @apply bg-light-600;
    }

    .col-type-content {
      @apply text-gray-warm;
    }

    .col-action {
      text-align: right;
    }

    .col-weight-content {
      text-align: center;
      @media (max-width: 1280px) {
        display: none;
      }
    }

    td {
      padding: 1.75rem 12px !important;
    }

    .col-key-content {
      display: inline-block;
      .username {
        display: flex;
        .user-img {
          margin: -2px 5px 0 0;
        }
      }
    }

    a:hover {
      text-decoration: underline;
    }
  }
}

.sign-dialog-header {
  display: flex;
  .sign-dialog-titles {
    margin-top: 9px;
  }
}

.success-dialog-header {
  display: flex;
  .success-dialog-titles {
    padding-top: 8px;
  }
}

.success-dialog-body {
  margin: 2rem 0 1.4rem 0;
  .success-dialog-content {
    text-align: center;
    margin-bottom: 1.4rem;
  }
}

.authority-main-title {
  font-weight: bold;
}
