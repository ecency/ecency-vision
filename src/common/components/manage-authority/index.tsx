import React, { useEffect, useState } from "react";
import { History } from "history";
import { PrivateKey } from "@hiveio/dhive";
import { ActiveUser } from "../../store/active-user/types";
import { Global } from "../../store/global/types";
import { AccountDataType, actionType } from "./types";
import UserAvatar from "../user-avatar/index";
import { error } from "../feedback";
import keyOrHot from "../key-or-hot";
import LinearProgress from "../linear-progress";
import ManageAuthIcon from "../manage-auth-icon";
import ManageKeys from "../manage-keys";
import { formatError, Revoke, RevokeHot, RevokeKc } from "../../api/operations";
import { getAccounts } from "../../api/hive";
import { _t } from "../../i18n";
import "./index.scss";
import { Modal, ModalBody, ModalHeader } from "@ui/modal";
import { Button } from "@ui/button";
import { Table, Td, Th, Tr } from "@ui/table";

interface Props {
  global: Global;
  activeUser: ActiveUser | null;
  signingKey: string;
  setSigningKey: (key: string) => void;
  history: History;
}

export default function ManageAuthorities(props: Props) {
  const [accountData, setAccountdata] = useState<AccountDataType>();
  const [newPostingsAuthority, setNewPostingsAuthority] = useState<[string, number][]>([]);
  const [step, setStep] = useState(0);
  const [keyDialog, setKeyDialog] = useState(false);
  const [targetAccount, setTargetAccount] = useState("");
  const [inProgress, setInProgress] = useState(false);

  useEffect(() => {
    getAccountData();
  }, []);

  const getAccountData = async () => {
    const response = await getAccounts([props.activeUser!.username]);
    if (response) {
      const resp = response[0];
      setAccountdata({
        postingsAuthority: resp.posting.account_auths,
        posting: resp.posting.key_auths[0],
        owner: resp.owner.key_auths[0],
        active: resp.active.key_auths[0],
        weight: resp.active.weight_threshold,
        memokey: resp.memo_key,
        account: response[0],
        publicKeys: {
          publicOwnerKey: resp.owner.key_auths[0][0],
          publicActiveKey: resp.active.key_auths[0][0],
          publicPostingKey: resp.posting.key_auths[0][0],
          publicMemoKey: resp.memo_key
        }
      });
    }
  };

  const toggleKeyDialog = () => {
    setKeyDialog(!keyDialog);
  };

  const handleRevoke = (account: string) => {
    setTargetAccount(account);
    setKeyDialog(true);
    setStep(1);
    setNewPostingsAuthority(accountData!.postingsAuthority.filter((x) => x[0] !== account));
  };

  const onKey = async (key: PrivateKey) => {
    try {
      setInProgress(true);
      const resp = await Revoke(
        props.activeUser!.username,
        accountData!.weight,
        newPostingsAuthority,
        [accountData!.posting],
        accountData!.memokey,
        key
      );
      if (resp.id) {
        setKeyDialog(true);
        setStep(2);
      }
    } catch (err) {
      error(...formatError(err));
    } finally {
      setInProgress(false);
    }
  };

  const onHot = () => {
    RevokeHot(
      props.activeUser!.username,
      accountData!.weight,
      newPostingsAuthority,
      [accountData!.posting],
      accountData!.memokey
    );
    setKeyDialog(false);
  };

  const onKc = () => {
    RevokeKc(
      props.activeUser!.username,
      accountData!.weight,
      newPostingsAuthority,
      [accountData!.posting],
      accountData!.memokey
    );
  };

  const finish = () => {
    setKeyDialog(false);
    getAccountData();
  };

  const signkeyModal = () => {
    return (
      <>
        <div className="sign-dialog-header border-b border-[--border-color]">
          <div className="step-no">1</div>
          <div className="sign-dialog-titles">
            <div className="authority-main-title">{_t("manage-authorities.sign-title")}</div>
            <div className="authority-sub-title">{_t("manage-authorities.sign-sub-title")}</div>
          </div>
        </div>
        {inProgress && <LinearProgress />}
        {keyOrHot({
          global: props.global,
          activeUser: props.activeUser,
          signingKey: props.signingKey,
          setSigningKey: props.setSigningKey,
          inProgress: inProgress,
          onKey: (key) => {
            onKey(key);
          },
          onHot: () => {
            toggleKeyDialog();
            if (onHot) {
              onHot();
            }
          },
          onKc: () => {
            toggleKeyDialog();
            if (onKc) {
              onKc();
            }
          }
        })}
        <p className="text-center">
          <a
            href="#"
            onClick={(e) => {
              e.preventDefault();
              setKeyDialog(false);
            }}
          >
            {_t("g.back")}
          </a>
        </p>
      </>
    );
  };

  const successModal = () => {
    return (
      <>
        <div className="success-dialog-header border-b border-[--border-color]">
          <div className="step-no">2</div>
          <div className="success-dialog-titles">
            <div className="authority-main-title">{_t("trx-common.success-title")}</div>
            <div className="authority-sub-title">{_t("trx-common.success-sub-title")}</div>
          </div>
        </div>

        <div className="success-dialog-body">
          <div className="success-dialog-content">
            <span>
              {" "}
              {_t("manage-authorities.success-message")}{" "}
              <a href={`https://ecency.com/@${targetAccount}`} target="_blank">
                {targetAccount}
              </a>{" "}
            </span>
          </div>
          <div className="flex justify-center">
            <span className="hr-6px-btn-spacer" />
            <Button onClick={finish}>{_t("g.finish")}</Button>
          </div>
        </div>
      </>
    );
  };

  const table = () => {
    return (
      <Table full={true}>
        <thead>
          <Tr>
            <Th>{_t("manage-authorities.type")}</Th>
            <Th>{_t("manage-authorities.key")}</Th>
            <Th className="hidden sm:table-cell" />
            <Th className="sm:hidden col-action">{_t("manage-authorities.actions")}</Th>
            <Th className="col-weight-content p-2">{_t("manage-authorities.weight")}</Th>
          </Tr>
        </thead>
        <tbody>
          {accountData!.postingsAuthority && accountData!.postingsAuthority.length > 0 && (
            <>
              {accountData!.postingsAuthority.map((account, i) => {
                return (
                  <>
                    <Tr key={i} className="tabl-row">
                      <Td className="col-type-content">{_t("manage-authorities.posting")}</Td>
                      {
                        <>
                          <Td>
                            <p className="col-key-content">
                              <a
                                className="username"
                                target="_blank"
                                href={`https://ecency.com/@${account[0]}`}
                              >
                                <span className="user-img">
                                  <UserAvatar username={account[0]} size="small" />
                                </span>

                                {account[0]}
                              </a>
                            </p>
                          </Td>
                          <Td className="hidden sm:table-cell">
                            {" "}
                            <Button onClick={() => handleRevoke(account[0])} outline={true}>
                              {_t("manage-authorities.revoke")}
                            </Button>
                          </Td>
                          <Td className="sm:hidden">
                            {
                              <ManageAuthIcon
                                history={props.history}
                                type={actionType.Revoke}
                                account={account[0]}
                                onRevoke={(account) => {
                                  handleRevoke(account);
                                }}
                              />
                            }
                          </Td>
                          <Td className="col-weight-content">{account[1]}</Td>
                        </>
                      }
                    </Tr>
                  </>
                );
              })}
            </>
          )}

          <ManageKeys
            history={props.history}
            accountData={accountData!}
            activeUser={props.activeUser!}
          />
        </tbody>
      </Table>
    );
  };

  return (
    <>
      <div className="authority-table">{accountData && table()}</div>

      {keyDialog && (
        <Modal
          animation={false}
          show={true}
          centered={true}
          onHide={toggleKeyDialog}
          className="authorities-dialog"
          size="lg"
        >
          <ModalHeader closeButton={true} />
          <ModalBody>
            {step === 1 && signkeyModal()}
            {step === 2 && successModal()}
          </ModalBody>
        </Modal>
      )}
    </>
  );
}
