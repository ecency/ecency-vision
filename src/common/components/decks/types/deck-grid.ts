import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  key: string;
  title: string;
  icon?: string;
  columns: DeckGridItem[];
  storageType: "account" | "local";
}

export interface DeckGrids {
  decks: DeckGrid[];
}
