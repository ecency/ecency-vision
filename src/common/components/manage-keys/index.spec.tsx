import React from "react";
import ManageKeys from "./index";
import renderer from "react-test-renderer";
import { createBrowserHistory } from "history";
import { activeUserInstance, fullAccountInstance } from "../../helper/test-helper";
import { AccountDataType } from "../manage-authority/types";

it("(1) Default render", () => {
  const postingsAuthority: AccountDataType["postingsAuthority"] = [
    ["ecency.app", 1],
    ["esteem-app", 1],
    ["esteem.app", 1],
    ["esteemapp", 1],
    ["peakd.app", 1]
  ];
  const posting: AccountDataType["posting"] = [
    "STM6uvU7j624wCZNa2pcXvkqmbnNC1cgraDrVG3pFRhXvj9LYS7Xp",
    1
  ];
  const owner: AccountDataType["owner"] = [
    "STM7F7zfd6ieangxz6uxQkYifUS5H841x5E41SYZaVc9F9cGPJ9jN",
    1
  ];
  const active: AccountDataType["active"] = [
    "STM71z4rmzGHdp7pmePZyS1G2GrbuBqg9PPHRgiK6uYqjzUTRNCRR",
    1
  ];
  const props = {
    accountData: {
      postingsAuthority: postingsAuthority,
      posting: posting,
      owner: owner,
      active: active,
      weight: 100,
      memokey: "STM7rioGL7NopT2Zo446hnzAiHLp5sJ1gS55xu3NikmxfQnEznsue",
      account: {
        ...fullAccountInstance
      },
      publicKeys: {
        publicOwnerKey: "owner_key",
        publicActiveKey: "active_key",
        publicPostingKey: "posting_key",
        publicMemoKey: "memo_key"
      }
    },
    activeUser: { ...activeUserInstance },
    history: createBrowserHistory()
  };
  const component = renderer.create(<ManageKeys {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
