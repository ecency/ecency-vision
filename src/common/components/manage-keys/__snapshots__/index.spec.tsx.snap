// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
Array [
  <tr
    className="[&:last-child>td]:border-b-0 [&:nth-child(even)]:bg-light-200 dark:[&:nth-child(even)]:bg-dark-300"
  >
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-type-content"
    >
       
      owner
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 key"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 hidden sm:table-cell"
    >
      <p
        className="action-btns"
      >
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 copy-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Copy
          </div>
        </button>
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 import-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Import private key
          </div>
        </button>
      </p>
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 sm:hidden"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-weight-content"
    >
      1
    </td>
  </tr>,
  <tr
    className="[&:last-child>td]:border-b-0 [&:nth-child(even)]:bg-light-200 dark:[&:nth-child(even)]:bg-dark-300"
  >
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-type-content"
    >
       
      active
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 key"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 hidden sm:table-cell"
    >
      <p
        className="action-btns"
      >
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 copy-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Copy
          </div>
        </button>
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 import-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Import private key
          </div>
        </button>
      </p>
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 sm:hidden"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-weight-content"
    >
      1
    </td>
  </tr>,
  <tr
    className="[&:last-child>td]:border-b-0 [&:nth-child(even)]:bg-light-200 dark:[&:nth-child(even)]:bg-dark-300"
  >
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-type-content"
    >
       
      posting
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 key"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 hidden sm:table-cell"
    >
      <p
        className="action-btns"
      >
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 copy-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Copy
          </div>
        </button>
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 import-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Import private key
          </div>
        </button>
      </p>
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 sm:hidden"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-weight-content"
    >
      1
    </td>
  </tr>,
  <tr
    className="[&:last-child>td]:border-b-0 [&:nth-child(even)]:bg-light-200 dark:[&:nth-child(even)]:bg-dark-300"
  >
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-type-content"
    >
       
      memo
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 key"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 hidden sm:table-cell"
    >
      <p
        className="action-btns"
      >
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 copy-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Copy
          </div>
        </button>
        <button
          className="cursor-pointer rounded-full duration-300 no-wrap border-[1.5px] border-solid border-blue-dark-sky hover:border-blue-dark-sky-hover focus:border-blue-dark-sky-active text-blue-dark-sky hover:text-blue-dark-sky-hover focus:text-blue-dark-sky-active h-[2.125rem] px-3 import-btn"
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
            }
          }
          type="button"
        >
          <div>
            Import private key
          </div>
        </button>
      </p>
    </td>
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 sm:hidden"
    />
    <td
      className="border-b border-r dark:border-gray-700 last:border-r-0 p-1 sm:p-2 col-weight-content"
    >
      1
    </td>
  </tr>,
]
`;
